base/pkg/entry.jl

```diff
@@ -50,19 +50,20 @@ function add(pkg::AbstractString, vers::VersionSet)
     outdated = :maybe
     @sync begin
         @async if !edit(Reqs.add,pkg,vers)
-            ispath(pkg) || throw(PkgError("unknown package $pkg"))
```

Comment:
No, it is not error. It's just happened that the required package is not registered with METADATA.